wit/metrics.wit

```diff
@@ -0,0 +1,272 @@
+interface metrics {
+    use wasi:clocks/wall-clock@0.2.0.{datetime};
+    use wasi:clocks/monotonic-clock@0.2.0.{duration};
+    use types.{key-value, instrumentation-scope};
+    use tracing.{span-id, trace-id};
+
+    /// `collect` gathers all metric data related to a Reader from the SDK
+    collect: func(metrics: resource-metrics) -> result<_, otel-error>;
+
+    /// `resource-metrics` is a collection of `scope-metrics` and the associated `resource`
+    /// that created them.
+    ///
+    /// See https://github.com/open-telemetry/opentelemetry-rust/blob/c811cde1ae21c624870c1b952190e687b16f76b8/opentelemetry-sdk/src/metrics/data/mod.rs#L13
+    record resource-metrics {
+        /// The entity that collected the metrics.
+        %resource: %resource,
+        /// The collection of metrics with unique `instrumentation-scope`s.
+        scope-metrics: list<scope-metrics>,
+    }
+
+    /// An immutable representation of the entity producing telemetry as attributes.
+    record %resource {
+        inner: resource-inner,
+    }
+
+    /// Inner structure of `resource` holding the actual data.
+    record resource-inner {
+        attributes: list<key-value>,
+        schema-url: option<string>,
+    }
```

Comment:
It's not obvious to me why we need `resource-inner`. My guess is this is a literal translation of what you saw in the Rust implementation but isn't a necessary pattern here in the WIT. I might be wrong though 🤷